ext/opcache/jit/zend_jit_ir.c

```diff
@@ -901,11 +922,17 @@ static ir_ref jit_IP(zend_jit_ctx *jit)
 	return ir_RLOAD_A(ZREG_IP);
 }
 
-static void jit_STORE_IP(zend_jit_ctx *jit, ir_ref ref)
+static void jit_STORE_IP_no_reset(zend_jit_ctx *jit, ir_ref ref)
 {
 	ir_RSTORE(ZREG_IP, ref);
 }
 
+static void jit_STORE_IP(zend_jit_ctx *jit, ir_ref ref)
+{
+	jit_STORE_IP_no_reset(jit, ref);
+	zend_jit_reset_last_valid_opline(jit);
```

Comment:
We must forget the last valid opline here by default, otherwise `zend_jit_set_ip()` will use a wrong offset. This used to be done in `zend_jit_reuse_ip()`.